src/coreclr/src/vm/mlinfo.cpp

```diff
@@ -2784,22 +2784,36 @@ MarshalInfo::MarshalInfo(Module* pModule,
                 // * Vector64<T>: Represents the __m64 ABI primitive which requires currently unimplemented handling
                 // * Vector128<T>: Represents the __m128 ABI primitive which requires currently unimplemented handling
                 // * Vector256<T>: Represents the __m256 ABI primitive which requires currently unimplemented handling
-                // * Vector<T>: Has a variable size (either __m128 or __m256) and isn't readily usable for inteorp scenarios
-
-                if (m_pMT->HasInstantiation() && (!m_pMT->IsBlittable()
-                    || m_pMT->HasSameTypeDefAs(g_pByReferenceClass)
-                    || m_pMT->HasSameTypeDefAs(g_pNullableClass)
-                    || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR64T))
-                    || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR128T))
-                    || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR256T))
+                // * Vector<T>: Has a variable size (either __m128 or __m256) and isn't readily usable for interop scenarios
+
+                if (m_pMT->HasInstantiation() && !IsFieldScenario())
+                {
+                    if (!m_pMT->IsBlittable()
+                        || (m_pMT->HasSameTypeDefAs(g_pByReferenceClass)
+                            || m_pMT->HasSameTypeDefAs(g_pNullableClass)
+                            || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR64T))
```

Comment:
`Vector64/128/256<T>` need to be blocked even in the field scenario. It has special ABI handling for some architectures, even when it is a field in a struct (where it might qualify as a HVA).